weatherman.py

```diff
@@ -0,0 +1,66 @@
+
+from parsers.weather_parser import WeatherParser
+from calculators.weather_stats_calculator import WeatherStatsCalculator
+from reports.report_generator import WeatherReportGenerator
+
+from helpers import invalid_format_provided, validate_date
+
+
+class Weatherman:
+
+    def __init__(self, arguments):
+        self.arguments = arguments
+        self.directory_path = arguments[1]
+
+    def driver(self):
+        parser = WeatherParser(self.directory_path)
+        temprature_readings = parser.parse_files()
+
+        for i in range(2, len(self.arguments), 2):
+            valid, result = validate_date(self.arguments[i+1])
+            if not valid:
+                invalid_format_provided(result)
+                return
+
+            option = self.arguments[i]
+            year, month = result
+            if month and (option == '-a' or option == '-c'):
+                print(f"\nGenerating report for year {year} and month {month} with option {option}")
+                self._generate_report_for_month(year, month, temprature_readings, option)
+            elif year and option == '-e' and month is None:
+                print(f"\nGenerating report for year {year} with option {option}")
+                self._generate_report_for_year(year, temprature_readings)
+            else:
+                invalid_format_provided("Invalid combination of option and date.")
+                return
+
+    def _generate_report_for_year(self, year, temprature_readings):
+        report_generator = WeatherReportGenerator()
+        yearly_readings = [r for r in temprature_readings if r.recorded_date.year == year]
```

Comment:
```suggestion
        yearly_readings = [temprature_reading for temprature_reading in temprature_readings if r.recorded_date.year == year]
```

Try to avoid variables that are not descriptive and promote mental mapping.